monolithic/monolithic-layered/src/test/java/com/epam/training/microservices/monolithic/service/delivery/DeliveryServiceTest.java

```diff
@@ -0,0 +1,78 @@
+package com.epam.training.microservices.monolithic.service.delivery;
+
+import com.epam.training.microservices.monolithic.model.delivery.Delivery;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class DeliveryServiceTest {
+
+    @Mock
+    private DeliveryRepository deliveryRepository;
+
+    @InjectMocks
+    private DeliveryService deliveryService;
+
+    @DisplayName("Test getRepository method")
+    @Test
+    void shouldGetRepository() {
+        JpaRepository<Delivery, Long> expectedRepo = deliveryRepository;
+        JpaRepository<Delivery, Long> resultRepo = deliveryService.getRepository();
+
+        assertEquals(expectedRepo, resultRepo);
+    }
+
+    @DisplayName("Test save method")
+    @Test
+    void shouldSaveDelivery() {
+        Delivery delivery = new Delivery();
+        when(deliveryRepository.save(delivery)).thenReturn(delivery);
+
+        Delivery savedDelivery = deliveryService.save(delivery);
+
+        assertEquals(delivery, savedDelivery);
```

Comment:
Another good think to check is that the method was actually called, and here Mockito’s `verify` is an outstanding fit. Something like this: 

```java
verify(deliveryRepository, times(1)).save(eq(delivery));
```

This is a useful thing to add.